core/blockchain.go

```diff
@@ -2979,22 +2980,44 @@ func (bc *BlockChain) GetExternalBlock(hash common.Hash, location []byte, contex
 }
 
 // requestExternalBlock sends an external block event to the missingExternalBlockFeed in order to be fulfilled by a manager or client.
-func (bc *BlockChain) requestExternalBlock(hash common.Hash, location []byte, context uint64) *types.ExternalBlock {
-	bc.missingExternalBlockFeed.Send(MissingExternalBlock{Hash: hash, Location: location, Context: int(context)})
-	for i := 0; i < params.ExternalBlockLookupLimit; i++ {
-		time.Sleep(time.Duration(params.ExternalBlockLookupDelay) * time.Millisecond)
-		// Lookup block in externalBlocks cache
-		key := types.ExtBlockCacheKey(context, hash)
-		if block, ok := bc.externalBlocks.HasGet(nil, key); ok {
-			var blockDecoded *types.ExternalBlock
-			rlp.DecodeBytes(block, &blockDecoded)
-			return blockDecoded
+func (bc *BlockChain) requestExternalBlock(hash common.Hash, blockContext uint64) *types.ExternalBlock {
```

Comment:
(nit) it might be better to ask subs first, since they are more lively. Specifically the sub in the originating slice (if we have that sub)